src/polynomial.cpp

```diff
@@ -0,0 +1,35 @@
+#include "polynomial.h"
+#include "add.h"
+
+namespace SymEngine {
+
+    Polynomial::Polynomial(const std::string& var, map_uint_mpz&& dict) :
```

Comment:
Did you benchmark it? If I remember well, unordered_map was a lot faster.
Make sure we can switch these two, so that we can benchmark it later and
switch if needed.

Sent from my mobile phone.
On May 24, 2015 8:13 AM, "Shivam Vats" <notifications@github.com> wrote:

> In src/polynomial.cpp
> <https://github.com/sympy/symengine/pull/454#discussion_r30952739>:
>
> > @@ -0,0 +1,35 @@
> > +#include "polynomial.h"
> > +#include "add.h"
> > +
> > +namespace SymEngine {
> > +
> > +    Polynomial::Polynomial(const std::string& var, map_uint_mpz&& dict) :
>
> Having numbers sorted usually speeds up computations. You might want to
> look at this
> <http://stackoverflow.com/questions/2196995/is-there-any-advantage-of-using-map-over-unordered-map-in-case-of-trivial-keys>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/symengine/pull/454/files#r30952739>.
>